### PR TITLE
Add DB init code location recommendation

### PR DESCRIPTION
**Pull request by scottaddie**
Fixes https://github.com/aspnet/Docs/issues/4134

This PR also renames sample app namespaces to fix VS warnings.

@divega Please review the newly-added **Move database initialization code** section:
[Internal Review Page](https://review.docs.microsoft.com/aspnet/core/migration/1x-to-2x/?branch=pr-en-us-4253#move-database-initialization-code)

### CODE REVIEW

File: aspnetcore/migration/1x-to-2x/index.md [hunk 1]
```diff
@@ -69,36 +69,54 @@ In the *.csproj* file, update the `Version` attribute of each `<DotNetCliToolRef
...
 
+<a name="db-init-code"></a>
+
+## Move database initialization code
+In 1.x projects using EF Core 1.x, a command such as `dotnet ef migrations add` does the following:
+1. Instantiates a `Startup` instance
+2. Invokes the `ConfigureServices` method to register all services with dependency injection (including `DbContext` types)
+3. Performs its requisite tasks
+
+In 2.0 projects using EF Core 2.0, the steps outlined above still occur. Additionally, the `Configure` method is invoked after `ConfigureServices`. If your 1.x app invoked database initialization code in its `Configure` method, unexpected problems can occur. For example, if the database doesn't yet exist, the seeding code will run before the EF Core Migrations command execution. This problem causes a `dotnet ef migrations list` command to fail if the database doesn't yet exist.
```
Comment (ajcvickers) [inline aspnetcore/migration/1x-to-2x/index.md]: ConfigureServices is not run in 2.0. /cc @bricelam for more detailed review on this.

Comment (ajcvickers) [reply aspnetcore/migration/1x-to-2x/index.md]: I think the cleanest way to think about this is that it is bad to do anything in BuildWebHost except to build and configure the web host. Anything that is about running the application should go outside of BuildWebHost--usually in Main. This allows tooling to get a WebHost without having side-effects related to starting the application.

File: aspnetcore/migration/1x-to-2x/samples/AspNetCoreDotNetCore1App/AspNetCoreDotNetCore1App/Controllers/MoviesController.cs [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+        // GET: Movies/Details/5
+        public async Task<IActionResult> Details(int? id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+
+            var movie = await _context.Movie
+                .SingleOrDefaultAsync(m => m.ID == id);
```
Comment (ajcvickers) [inline aspnetcore/migration/1x-to-2x/samples/AspNetCoreDotNetCore1App/AspNetCoreDotNetCore1App/Controllers/MoviesController.cs]: Consider using Find.

File: aspnetcore/migration/1x-to-2x/samples/AspNetCoreDotNetCore1App/AspNetCoreDotNetCore1App/Controllers/MoviesController.cs [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+        // GET: Movies/Edit/5
+        public async Task<IActionResult> Edit(int? id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+
+            var movie = await _context.Movie.SingleOrDefaultAsync(m => m.ID == id);
```
Comment (ajcvickers) [inline aspnetcore/migration/1x-to-2x/samples/AspNetCoreDotNetCore1App/AspNetCoreDotNetCore1App/Controllers/MoviesController.cs]: Consider using Find.

File: aspnetcore/migration/1x-to-2x/index.md (hunk 1 above)
Comment (bricelam) [reply aspnetcore/migration/1x-to-2x/index.md]: > In 2.0 projects using EF Core 2.0, the steps outlined above still occur. Additionally, the `Configure` method is invoked after `ConfigureServices`.

More like: In 2.0 projects using EF Core 2.0, `Program.BuildWebHost` is invoked to obtain the application services. Unlike 1.x, this has the additional side affect of invoking `Startup.Configure`.

(But I'm also OK over-simplifying if it makes it easier to understand.)

Comment (bricelam) [reply aspnetcore/migration/1x-to-2x/index.md]: I might also explicitly call out that the semantics of `Startup.Configure` are to configure the request pipeline, so using it as an app initialization hook is not correct.

**Pull request merged**